python/location/lbs/activity/audioset/dataprocessing/audio_processing.py

```diff
@@ -0,0 +1,262 @@
+# This Python file is an extended/modified version of Jerry Wu's audio_prep.py
+# This python file is a script to download a given label from AudioSet
+# The downloaded audios would be clipped into 10 seconds periods
+# The downloaded audio segments are then processed using Librosa to extract features
+# Stored in eval_segments, balanced_train, and unbalanced_train separately.
+
+# Usage: python3 audio_prep.py <label>
+
+
+from __future__ import unicode_literals
+import youtube_dl
+import csv
+import json
+from pydub import AudioSegment
+import os
+import shutil
+from os import listdir
+from os.path import isfile, join
+import sys
+import numpy as np
+import librosa
+import argparse
+import numpy as np
+
+file_structure = "python/location/lbs/activity/audioset/dataprocessing/"
+referDoc = "ReferDoc/"
+downloaded_audio = []
+feature_dict = {}
+
+
+def setup_commandline_args():
+  parser = argparse.ArgumentParser(description='Process dataset.csv to input into TensorFlow.')
+  parser.add_argument('-d', '--datasets', required=True, type=str, nargs='+')
```

Comment:
haven't tried it myself but the following looks like an external google library you can use for flags:
https://abseil.io/docs/python/guides/flags